pkg/extensions/sync/service.go

```diff
@@ -479,11 +554,78 @@ func (service *BaseService) syncRef(ctx context.Context, localRepo string, remot
 		service.log.Info().Str("remote image", remoteImageRef.CommonName()).
 			Str("local image", fmt.Sprintf("%s:%s", localRepo, remoteImageRef.Tag)).Msg("syncing image")
 
-		err = service.rc.ImageCopy(ctx, remoteImageRef, localImageRef, copyOpts...)
-		if err != nil {
-			service.log.Error().Err(err).Str("errortype", common.TypeOf(err)).
-				Str("remote image", remoteImageRef.CommonName()).
-				Str("local image", fmt.Sprintf("%s:%s", localRepo, remoteImageRef.Tag)).Msg("failed to sync image")
+		// When platforms are specified, we need to filter the manifest list
+		if len(service.config.Platforms) > 0 {
+			// Get the manifest to check if it's a manifest list
+			man, err := service.rc.ManifestGet(ctx, remoteImageRef)
+			if err != nil {
+				service.log.Error().Err(err).Str("errortype", common.TypeOf(err)).
+					Str("remote image", remoteImageRef.CommonName()).
+					Msg("failed to get manifest for architecture filtering")
+				return err
+			}
+
+			// If it's a manifest list (multi-arch image), we need to filter architectures
+			_, isIndexer := man.(manifest.Indexer)
+			if isIndexer {
+				service.log.Info().
+					Str("remote image", remoteImageRef.CommonName()).
+					Msg("filtering platforms for multi-arch image")
+
+				// Use ImageCopy with the architecture filtering options
+				err = service.rc.ImageCopy(ctx, remoteImageRef, localImageRef, copyOpts...)
+				if err != nil {
+					service.log.Error().Err(err).Str("errortype", common.TypeOf(err)).
+						Str("remote image", remoteImageRef.CommonName()).
+						Str("local image", fmt.Sprintf("%s:%s", localRepo, remoteImageRef.Tag)).
+						Msg("failed to sync image")
+				}
+
+				// The architecture filtering will be applied during processing before the commit stage
+				// The actual filtering happens in the destination.CommitAll method
+			} else {
+				// It's a single-arch image, verify if we should include this platform
+				if man.GetDescriptor().Platform != nil {
+					// Convert platform to ispec.Platform for compatibility with regclient v0.8.3
+					platform := &ispec.Platform{
+						Architecture: man.GetDescriptor().Platform.Architecture,
+						OS:           man.GetDescriptor().Platform.OS,
+						OSVersion:    man.GetDescriptor().Platform.OSVersion,
+						OSFeatures:   man.GetDescriptor().Platform.OSFeatures,
+						Variant:      man.GetDescriptor().Platform.Variant,
+					}
+					if !service.shouldIncludePlatform(platform) {
+						platformDesc := platform.Architecture
+						if platform.OS != "" {
+							platformDesc = platform.OS + "/" + platform.Architecture
```

Comment:
1. Do we need to handle variant here?
2. Maybe we should have a Platform method to return the os/arch/variant string for that instance.